_posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc

```diff
@@ -0,0 +1,549 @@
+core module] creates a best practices set of IAM groups. The groups are configurable according to your needs.
+You could, for example, opt to create a group with read only access, another group with full administrator
+access, and no other groups. All Gruntwork subscribers have access to the core modules, which reside in
+Gruntwork's infrastructure as code repositories.
+
+[[wrapper_modules]]
+=== Compliance wrapper modules
+The compliance wrapper modules are an extension of the IaC Library. They use the
+link:https://www.terraform.io/docs/modules/sources.html[`source` argument in a Terraform module block] to call
+the core module with a configuration that is customized for compliance with the CIS AWS Foundations Benchmark.
+These modules are in the link:https://github.com/gruntwork-io/cis-compliance-aws[`cis-compliance-aws`
+repository] (accessible to Gruntwork Compliance subscribers).
+
+[[infrastructure_modules]]
+=== infrastructure-modules
+The `infrastructure-modules` are your organization's "blueprint" for how to deploy infrastructure. You can
+use `infrastructure-modules` to customize the settings according to the needs of your environment. Subscribers
+can refer to the
+canonical link:https://github.com/gruntwork-io/infrastructure-modules-multi-account-acme[ACME
+infrastructure-modules reposistory].
+
+[[infrastructure_live]]
+=== infrastructure-live
+`infrastructure-live` uses link:https://github.com/gruntwork-io/terragrunt[Terragrunt] to make it easier to
+work with Terraform modules in multiple environments. `infrastructure-live` is optional - you can use all of
+the modules with or without Terragrunt. Refer to the canonical
+link:https://github.com/gruntwork-io/infrastructure-live-multi-account-acme[ACME infrastructure-live
+repository], and in particular the
+link:https://github.com/gruntwork-io/infrastructure-live-multi-account-acme/tree/master/_docs[Infrastructure
+walkthrough] for comprehensive documentation on how `infrastructure-live`, `infrastructure-modules`, and the
+core IaC modules interact.
+ +
+[[benefits]]
+=== Benefits
+This modular, decomposed approach allows for maximum code reuse. The core modules can be used with or without
+compliance, depending on how they are configured. The compliance wrappers are like shadows of the core
+modules; they pass through most of the variables to the core modules without alteration, but hard code any
+settings needed for compliance. When you call the compliance modules from your own code in
+`infrastructure-modules`, you only need to set up any variables that are custom for your environment. Often
+times the default settings are good enough.
+ +
+ +
+
+If you need to brush up on how the IaC Library works, read the
+link:https://gruntwork.io/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library/[How to use
+the Gruntwork Infrastructure as Code Library] guide.
+
+[[deployment_walkthrough]]
+== Deployment walkthrough
```

Comment:
This is a good overview, but I think the intention of these guides is that they should be even more detailed than this, down to the folder structure + specific code samples. See this section of the VPC guide for example: https://gruntwork.io/guides/networking/how-to-deploy-production-grade-vpc-aws/#deploy_mgmt_vpc

Reply:
[Added some code examples](https://github.com/gruntwork-io/gruntwork-io.github.io/pull/225/commits/bb254c13d2ad1451cb572a9836d1e2422efc62ef).

Reply:
I added code examples in the section "Using the wrapper module". The trick here is not to clutter the guide with TOO much code such that it becomes overwhelming. After all, the terragrunt and terraform examples within the repositories have all the code for each module. The balance I've tried to strike here is to give one example - the IAM password policy wrapper module, both for terraform and terragrunt - and instruct users to follow the same pattern for all the wrapper modules.

Reply:
I think that approach is reasonable!